documentation/misc/Building_a_network_skeleton.md

```diff
@@ -0,0 +1,220 @@
+        workerCount := MigrationMaxWorkerCount
+        if workerCount <= 4 {
+            workerCount = 1
+        } else {
+            workerCount /= 2
+        }
+
+        lbts, lbRoot, err := stmgr.GetLookbackTipSetForRound(ctx, sm, ts, epoch)
+        if err != nil {
+            return xerrors.Errorf("error getting lookback ts for premigration: %w", err)
+        }
+
+        config := migration.Config{
+            MaxWorkers:        uint(workerCount),
+            ProgressLogPeriod: time.Minute * 5,
+        }
+
+        _, err = upgradeActorsV(XX+1)Common(ctx, sm, cache, lbRoot, epoch, lbts, config)
+        return err
+    }
+
+    func UpgradeActorsV(XX+1)(ctx context.Context, sm *stmgr.StateManager, cache stmgr.MigrationCache, cb stmgr.ExecMonitor,
+        root cid.Cid, epoch abi.ChainEpoch, ts *types.TipSet) (cid.Cid, error) {
+        // Use all the CPUs except 2.
+        workerCount := MigrationMaxWorkerCount - 3
+        if workerCount <= 0 {
+            workerCount = 1
+        }
+        config := migration.Config{
+            MaxWorkers:        uint(workerCount),
+            JobQueueSize:      1000,
+            ResultQueueSize:   100,
+            ProgressLogPeriod: 10 * time.Second,
+        }
+        newRoot, err := upgradeActorsV(XX+1)Common(ctx, sm, cache, root, epoch, ts, config)
+        if err != nil {
+            return cid.Undef, xerrors.Errorf("migrating actors v11 state: %w", err)
+        }
+        return newRoot, nil
+    }
+
+    func upgradeActorsV(XX+1)Common(
+        ctx context.Context, sm *stmgr.StateManager, cache stmgr.MigrationCache,
+        root cid.Cid, epoch abi.ChainEpoch, ts *types.TipSet,
+        config migration.Config,
+    ) (cid.Cid, error) {
+        writeStore := blockstore.NewAutobatch(ctx, sm.ChainStore().StateBlockstore(), units.GiB/4)
+        adtStore := store.ActorStore(ctx, writeStore)
+        // ensure that the manifest is loaded in the blockstore
+        if err := bundle.LoadBundles(ctx, writeStore, actorstypes.Version(XX+1)); err != nil {
+            return cid.Undef, xerrors.Errorf("failed to load manifest bundle: %w", err)
+        }
+
+        // Load the state root.
+        var stateRoot types.StateRoot
+        if err := adtStore.Get(ctx, root, &stateRoot); err != nil {
+            return cid.Undef, xerrors.Errorf("failed to decode state root: %w", err)
+        }
+
+        if stateRoot.Version != types.StateTreeVersion5 {
+            return cid.Undef, xerrors.Errorf(
+                "expected state root version 5 for actors v(XX+1) upgrade, got %d",
+                stateRoot.Version,
+            )
+        }
+
+        manifest, ok := actors.GetManifest(actorstypes.Version(XX+1))
+        if !ok {
+            return cid.Undef, xerrors.Errorf("no manifest CID for v(XX+1) upgrade")
+        }
+
+        // Perform the migration
+        newHamtRoot, err := nv(XX+1).MigrateStateTree(ctx, adtStore, manifest, stateRoot.Actors, epoch, config,
```

Comment:
This assumes that you've already created a `MigrateStateTree` function in go-state-types, but that item is missing from the go-state-types checklist.

Reply:
I think it would be good to add a similar doc for go-state-types that includes a simple migration that you can copy. Thanks to some recent refactors, it's pretty easy to create the "base migration". I'm happy to help produce that.

Reply:
> I think it would be good to add a similar doc for go-state-types that includes a simple migration that you can copy. Thanks to some recent refactors, it's pretty easy to create the "base migration". I'm happy to help produce that.

Agreed! Are there any commits/PRs that you can point me to, where I can look how on create a base migration? For the documentation in go-state-types, would it be most convenient to create a new `Building_actor_version_skeleton` documentation file, or expand on the current actor_version_checklist: https://github.com/filecoin-project/go-state-types/pull/192

Reply:
Discussed on Slack, but please let me know if you'd like any more pointers!

I would personally suggest just expanding the existing actor_version_checklist in go-state-types, but it's up to you!